### DISCUSSION TITLE
Filtering addresses is not working

### DISCUSSION THREAD

**Discussion by jorgegb7** [evilsocket/opensnitch#573] [Q&A]
Hello, first of all I want to say thank you for this awesome project and the effort to keep it always up to date.

I was trying to block few websites like youtube.com or facebook.com but allow all the other connections. So what I do is create .txt file with a list of the addresses that I want to block following the same pattern:
0.0.0.0 youtube.com
0.0.0.0 facebook.com

I try to set up the same way is in the github wiki but it won't work. I go to youtube.com but it will load without any problem. Is there anything in the configuration that I need to do in order to make this blocks work?

I have tried also following the tutorial on the wiki to block ads but still not working. I will leave attached couple screenshots with the config and the rules.

Thank you,
Jorge.

![1](https://user-images.githubusercontent.com/43822937/147135664-22e4dc61-4f51-4f04-b957-0e501e014622.png)

**Comment by gustavo-iniguez-goya** [comment]
Hi Jorge!

What version are you using? v1.4.3? 

Name also the rule as `000-test`, just in case there's another rule that is allowing it. Have you checked out if any of the other rules is being allowing the connection?

On the other hand, sometimes web browsers despite you type `domain.com`, they redirect you to `www.domain.com`. Try adding also `www.youtube.com`.

**Reply by jorgegb7** [reply]
Hello Gustavo,

I am using 1.4.2, 

As I am seeing the problem could be because when I start up the system I allow until next reboot connections to my ISP, it always ask for it when I open the browser. Should I create a rule that allows all the connections to avoid the firewall pop up all the time on start and then another rule that blocks the connection to this websites?

Seems for what you comment that probably that rule is allowing all the connections and jumping the restrictions that I set in my rule, could be that the problem?

Thank you!

**Reply by gustavo-iniguez-goya** [reply]
> when I start up the system I allow until next reboot connections to my ISP ... Should I create a rule that allows all the connections to avoid the firewall pop up all the time on start and then another rule that blocks the connection to this websites?

Yes, I think so. You can create the rules as follow:

000-deny-list -> [x] Precedence, [x] Enable [x] Deny [x] List: /tmp/test/
allow-traffic -> [x] Enable [x] Allow

If you double click on a rule, you'll see all the connections that it has intercepted, that way you can see if a particular domain has been allowed/denied by this or that rule.

**Reply by jorgegb7** [reply]
I've been trying and no success, right now the rules are like this:
000-deny-list -> [x] Priority rule, [x] Enable [x] Deny [x] List: /tmp/test/
allow-traffic -> [x] Enable [x] Allow [x] From this executable: /usr/lib/firefox-esr/firefox-esr
Allow traffic made me mark at least one characteristic else, so I choosed to add all the traffic generated by my browser.

No other rule has been created, with that configuration should be working, right ? It still does not block me the traffic to the websites on my list. The list is a .txt, could be that the problem? 

Thank you!

**Reply by gustavo-iniguez-goya** [reply]
oops, sorry @jorgegb7 , I missed this comment (it didn't generate a notification).

With that configuration it should work. Could you set log level to DEBUG (under Preferences -> Nodes), and paste the relevant lines of /var/log/opensnitchd.log ?

Like:
- set log level to DEBUG 
- empty log: `truncate -s 0 /var/log/opensnitchd.log`
- open firefox and load a domain present in your .txt file
- copy .log to other location and post it here: `cp /var/log/opensnitchd.log /tmp/`

*.txt files should load just fine.

Oh by the way, there should be logs like this one in `opensnitchd.log`:
```
Loading domains list: /tmp/test/test.txt, size: 33
13 domains loaded, /tmp/test/test.txt
1 lists loaded, 13 domains, 0 duplicated
```

**Reply by jorgegb7** [reply]
It finally worked, I think I probably messed it up with the order of the rules! Thank you so much!
